### ISSUE TITLE
Lint code fences in markdown files?

### ISSUE THREAD

**Issue by tunnckoCore** [AtomLinter/linter-eslint#779]
## Issue Type

Question / Bug

## Issue Description
I'm not sure if the issue is for here, actually.

I have `.eslintrc`

```json
{
  "extends": ["standard"],
  "plugins": ["markdown"]
}
```

and have just `linter` and `linter-eslint` packages. When run eslint from CLI it reports them as expected. But when typing in the markdown file it does not report any errors as like in the javascript files. So it seems it does not detects or respects the `eslint-plugin-markdown`?

I also added `source.md` in the `linter-eslint` settings.

## Bug Checklist
- Restart Atom - **yes**
- Verify the `eslint` CLI gives the proper result, while `linter-eslint` does not - **yes**
- Paste the output of the `Linter Eslint: Debug` command from the Command Palette below

```

atom version: 1.12.9
linter-eslint version: 8.0.0
eslint version: 3.13.1
hours since last atom restart: 0.3
platform: linux
Using local project eslint from /home/charlike/hub/fasfsdfsdf/node_modules/eslint
linter-eslint configuration: {
  "fixOnSave": true,
  "globalNodePath": "/home/charlike/.nvm/versions/node/v7.3.0",
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic",
    "source.md",
    "source.markdown"
  ],
  "lintHtmlFiles": false,
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDir": "",
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "rulesToSilenceWhileTyping": []
}
```

the tested markdown file

```md

## Foobar

Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

\\`\`\`js
const foo = 1;

foo = 123
\`\`\`

Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

```

eslint cli

```
~/hub/fasfsdfsdf  
% 1 ❯ eslint --ext md .                                                                                 January 11, 01:43:40

/home/charlike/hub/fasfsdfsdf/foo.md
  7:7   error  'foo' is assigned a value but never used  no-unused-vars
  7:14  error  Extra semicolon                           semi
  9:1   error  'foo' is constant                         no-const-assign

✖ 3 problems (3 errors, 0 warnings)

```

**Comment by tunnckoCore** [comment]
Both locally and globally installed

```
~/hub/fasfsdfsdf  
% ❯ npm ls | grep eslint                                                                                January 11, 02:01:52
├─┬ eslint@3.13.1
├── eslint-config-standard@6.2.1
├─┬ eslint-plugin-markdown@1.0.0-beta.3
├── eslint-plugin-promise@3.4.0
└── eslint-plugin-standard@2.0.1
```

**Comment by Arcanemagus** [comment]
This should work just fine if you add the right scope, the default Markdown language in Atom marks Markdown files as `source.gfm` not `source.md`.

Another thing to note is that the blocks need to be "real" embedded blocks, your example has the \` commented out so it reads as plain Markdown text.

Tested with the following files: [linter-eslint_GH779.zip](https://github.com/AtomLinter/linter-eslint/files/767921/linter-eslint_GH779.zip)

**Comment by tunnckoCore** [comment]
@Arcanemagus i commented tout hem because can't be shown correctly here at the issue. They are real.

**Comment by Arcanemagus** [comment]
I figured, since your CLI results would be impossible otherwise 😛.

Does this work properly for you if you change the scope to `source.gfm`?

**Comment by tunnckoCore** [comment]
I'll try.

**Comment by tunnckoCore** [comment]
Hm. I have `language-gfm` enabled? And the md file is using `GitHub Markdown`. For js files it seems to work, intentionally set to insert semi colons so i can see the diff. When I'm in markdown file in seems like it now insert them, nor shows any errors.

A bit later. Ooh, okey I set the `source.gfm` in the `linter-eslint` settings, the debug info seems that it includes it.

```json
"scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic",
    "source.md",
    "source.gfm",
    "source.markdown"
  ],
```

but it shows me an error ` Parsing error: Unexpected character '#'at line 1 col 1` when i'm in README.md

**Comment by tunnckoCore** [comment]
I also tried to remove the others `source.md` and `source.markdown` it still fails.

**Comment by tunnckoCore** [comment]
Doh! My fault. Didn't include the markdown plugin in the config, no i see the correct style errors like `unused var` and etc.  

But there is another issue. It seems it not auto fix the errors? I'll open another issue.
